### PR TITLE
CI: Use build-push-action v2.

### PR DESCRIPTION
**Pull request by csadorf**
Closes issue #117.

### CODE REVIEW

**Comment by csadorf** [comment]
@yakutovicha This should significantly speed up build and test times for this repository (~8mins vs ~25mins).

**Comment by yakutovicha** [comment]
@csadorf what about your comment on the experimental status of the action?

**Comment by csadorf** [comment]
> @csadorf what about your comment on the experimental status of the action?

The action's API in v2 appears to have been rather stable for a while and the v2 beta[ is officially announced](https://github.com/docker/build-push-action/commit/35323d9aa4928042a3a3317ed0fe198e051f7df4). I believe the benefits (massively sped up builds) outweigh a potential necessity to slightly update this workflow should there be unexpected changes.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,25 +15,52 @@ jobs:
...
-      image: ${{ steps.tag.outputs.image }}
+      tag: ${{ steps.setup.outputs.tag }}
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: build image
-        uses: docker/build-push-action@v1
+      -
+        uses: actions/checkout@v2
```
Comment (yakutovicha) [inline .github/workflows/ci.yml:22]: Maybe, for the consistency, use this style everywhere.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,25 +15,52 @@ jobs:
...
+      -
+        name: Set image tag
+        id: setup
+        run: |
+          IMAGE_REF=`echo ${{ github.sha }} | sed 's/^\(.......\).*/\1/'`
+          TAG="sha-${IMAGE_REF}"
+          echo "::set-output name=tag::sha-${IMAGE_REF}"
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (yakutovicha) [inline .github/workflows/ci.yml:32]: If I am not mistaking this is supposed to test docker on different architectures, right? Is it actually happening?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,25 +15,52 @@ jobs:
...
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (yakutovicha) [inline .github/workflows/ci.yml:35]: This thing is supposed to provide access to the early/still-in-development features of the Docker CLI? Do we really need them? Which experimental features exactly are we using?

**Review by yakutovicha — COMMENTED** [review]
Thanks, @csadorf. Just a few questions/comments?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (csadorf) [reply .github/workflows/ci.yml:32]: I just followed the example here: https://github.com/docker/build-push-action/tree/master#git-context

File: .github/workflows/ci.yml (hunk 3 above)
Comment (csadorf) [reply .github/workflows/ci.yml:35]: I just followed the example here: https://github.com/docker/build-push-action/tree/master#git-context

File: .github/workflows/ci.yml (hunk 1 above)
Comment (csadorf) [reply .github/workflows/ci.yml:22]: sure

**Comment by csadorf** [comment]
While the travis workflow succeeds, the GA actions workflow fails when either the buildx and/or the qemu action is removed:
- https://github.com/csadorf/aiidalab-docker-stack/runs/1224882260?check_suite_focus=true
- https://github.com/csadorf/aiidalab-docker-stack/runs/1224884057?check_suite_focus=true

**Comment by csadorf** [comment]
@yakutovicha Adopted style for whole workflow config file and demonstrated that the additional actions (buildx and qemu) are required for the workflow to succeed. Ready for review.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (csadorf) [reply .github/workflows/ci.yml:32]: This step is required, see comment below.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (csadorf) [reply .github/workflows/ci.yml:35]: This step is required, see comment below.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (yakutovicha) [reply .github/workflows/ci.yml:32]: ok

File: .github/workflows/ci.yml (hunk 3 above)
Comment (yakutovicha) [reply .github/workflows/ci.yml:35]: ok

**Review by yakutovicha — APPROVED** [review]

**Pull request merged**